packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -739,50 +730,32 @@ function recursivelyDisconnectChildren(vnodes: VNodes) {
 // into snabbdom. Especially useful when the reset is a consequence of an error, in which case the
 // children VNodes might not be representing the current state of the DOM.
 export function resetComponentRoot(vm: VM) {
-    const {
-        children,
-        renderRoot,
-        renderer: { remove },
-    } = vm;
-
-    for (let i = 0, len = children.length; i < len; i++) {
-        const child = children[i];
-
-        // VFragments are special; their .elm property does not point to the root element since they have no root,
-        // so we have to clean them up differently.
-        if (!isNull(child)) {
-            if (isVFragment(child)) {
-                removeFragmentChildren(child, vm);
-            } else if (!isUndefined(child.elm)) {
-                remove(child.elm, renderRoot);
-            }
-        }
-    }
+    recursivelyRemoveChildren(vm.children, vm);
```

Comment:
Love seeing so much red!

Reply:
It's probably because the logic is overall still the same for `if:true`. We went from:
```javascript
for (let i = 0, len = children.length; i < len; i++) {
    const child = children[i];

    if (!isNull(child) && !isUndefined(child.elm)) {
        remove(child.elm, renderRoot);
    }
}
```
to
```javascript
for (let i = 0, len = vnodes.length; i < len; i += 1) {
    const vnode = vnodes[i];

    if (!isNull(vnode)) {
        if (isVFragment(vnode)) {
            recursivelyRemoveChildren(vnode.children, vm);
        } else if (!isUndefined(vnode.elm)) {
            remove(vnode.elm, renderRoot);
        }
    }
}
```
No existing use cases _should_ go down the new recursive path we've added.